### DISCUSSION TITLE
how to set jwt_refresh_token_deadline_interval ?

### DISCUSSION THREAD

**Discussion by francescor** [jeremyevans/rodauth#388] [General]
Where should I customize the value of `jwt_refresh_token_deadline_interval` ?

I've tried 
```
jwt_refresh_token_deadline_interval 7
```

in app/misc/rodauth_main.rb but it generate an error

**Comment by janko** [comment]
You need to set a hash with keys support by [date_arithmetic](https://sequel.jeremyevans.net/rdoc-plugins/files/lib/sequel/extensions/date_arithmetic_rb.html) Sequel plugin, e.g:

```rb
jwt_refresh_token_deadline_interval Hash[days: 7]
```

It seems that date_arithmetic Sequel plugin also supports `ActiveSupport::Duration`, so the following might work as well:

```rb
jwt_refresh_token_deadline_interval 7.days
```

**Comment by francescor** [comment]
yes! thank you so much!

``` ruby
# app/misc/rodauth_main.rb
  
  require "sequel/core"
  class RodauthMain < Rodauth::Rails::Auth
  ...
  jwt_refresh_token_deadline_interval 7.days
  ...
```

I see no errors with the above: we still have to check the effects, but no errors.

Thank you, Janko, for your almost immediate great answer !
